### ISSUE TITLE
Memory mapped compressed feather files

### ISSUE THREAD

**Issue by thobai** [pola-rs/polars#5890]
### Problem description

I wish Polars would support memory mapping for compressed feather files in the same way it does for parquet. This would certainly improve reading performance of `scan_ipc` especially for subsequent data access where the OS has cached the memory mapped pages. This  should make this the fastest option to read / scan data for Polars as there is no conversion required from the layout on disk to the layout in memory.

In a nutshell, I wish `polars.scan_ipc(file = 'data.feather.lz4', memory_map = True)` would work (equally for zstd compression).

**Comment by arturdaraujo** [comment]
That's very interesting. I fully agree with this feature request

**Comment by ritchie46** [comment]
We already open the file memory mapped. I think the most performance benefit is in trying to parallelize the IPC reader but that needs some upstream arrow2 changes.

**Comment by jmakov** [comment]
@thobai as a workaround you can use filesystem based compression e.g. BTRFS with ZSTD-9.

**Comment by coastalwhite** [comment]
I think this is fixed now that you can give arbitrary byte sources to `scan_*` functions.
